pkg/pillar/cmd/volumemgr/create.go

```diff
@@ -22,46 +22,60 @@ func createVolume(ctx *volumemgrContext, status types.VolumeStatus) (bool, strin
 		log.Infof("createVolume(%s) from container %s", status.Key(), status.ReferenceName)
 		return createContainerVolume(ctx, status, status.ReferenceName)
 	}
-	log.Infof("createVolume(%s) from disk %s", status.Key(), status.FileLocation)
-	return createVdiskVolume(ctx, status, status.FileLocation)
+	log.Infof("createVolume(%s) from disk %s", status.Key(), status.ReferenceName)
+	return createVdiskVolume(ctx, status, status.ReferenceName)
 }
 
 // createVdiskVolume does not update status but returns
 // new values for VolumeCreated, FileLocation, and error
 func createVdiskVolume(ctx *volumemgrContext, status types.VolumeStatus,
-	srcLocation string) (bool, string, error) {
+	ref string) (bool, string, error) {
 
 	created := false
-	if status.ReadOnly {
-		log.Infof("createVolume(%s) ReadOnly", status.Key())
-		created = true // To make doUpdate proceed
-		return created, srcLocation, nil
-	}
 
+	// this is the target location, where we expect the volume to be
 	filelocation := status.PathName()
 	if _, err := os.Stat(filelocation); err == nil {
 		errStr := fmt.Sprintf("Can not create %s for %s: exists",
 			filelocation, status.Key())
 		log.Error(errStr)
-		return created, srcLocation, errors.New(errStr)
+		return created, ref, errors.New(errStr)
```

Comment:
Here (and below) you return ref which is not a pathname.

Reply:
Updated.